xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/keys/Seed.java

```diff
@@ -134,81 +157,329 @@ public static Seed secp256k1SeedFromEntropy(final Entropy entropy) {
+   * A default implementation of a {@link Seed}.
    */
-  @VisibleForTesting
-  Seed(final UnsignedByteArray value) {
-    this.value = Objects.requireNonNull(value);
-  }
+  class DefaultSeed implements Seed {
+
+    // This holds all Seed bytes (e.g., entropy bytes, version type, etc).
+    private final UnsignedByteArray value;
+    private boolean destroyed;
+
+    /**
+     * Required-args Constructor. Purposefully package-private for testing purposes only (use a static method instead
+     * for construction).
+     *
+     * @param value This seed's full binary value (including the entropy bytes and versionType).
+     */
+    @VisibleForTesting
+    DefaultSeed(final UnsignedByteArray value) {
+      this.value = Objects.requireNonNull(value);
+    }
 
-  /**
-   * Copy constructor.
-   *
-   * @param seed This seed's full binary value (including the entropy bytes and versionType).
-   */
-  @VisibleForTesting
-  Seed(final Seed seed) {
-    Objects.requireNonNull(seed);
-    this.value = UnsignedByteArray.of(seed.value.toByteArray());
-  }
+    /**
+     * Copy constructor.
+     *
+     * @param seed This seed's full binary value (including the entropy bytes and versionType).
+     */
+    @VisibleForTesting
+    DefaultSeed(final DefaultSeed seed) {
+      Objects.requireNonNull(seed);
+      this.value = UnsignedByteArray.of(seed.value.toByteArray());
+      this.destroyed = seed.isDestroyed();
+    }
 
-  /**
-   * The decoded details of this seed.
-   *
-   * @return An instance of {@link Decoded}.
-   */
-  public Decoded decodedSeed() {
-    final byte[] copiedByteValue = new byte[this.value.length()];
-    System.arraycopy(this.value.toByteArray(), 0, copiedByteValue, 0, value.length());
+    /**
+     * The decoded details of this seed.
+     *
+     * @return An instance of {@link Decoded}.
+     */
+    public Decoded decodedSeed() {
+      final byte[] copiedByteValue = new byte[this.value.length()];
+      System.arraycopy(this.value.toByteArray(), 0, copiedByteValue, 0, value.length());
+
+      return SeedCodec.getInstance().decodeSeed(
+        Base58.encode(copiedByteValue)
+      );
+    }
 
-    return SeedCodec.getInstance().decodeSeed(
-      Base58.encode(copiedByteValue)
-    );
-  }
+    @Override
+    public KeyPair deriveKeyPair() {
+      VersionType type = this.decodedSeed().type()
+        .orElseThrow(() -> new IllegalArgumentException("Unsupported seed type."));
+
+      switch (type) {
+        case ED25519: {
+          return Ed25519KeyPairService.deriveKeyPair(this);
+        }
+        case SECP256K1: {
+          return Secp256k1KeyPairService.deriveKeyPair(this);
+        }
+        default: {
+          throw new IllegalArgumentException("Unsupported seed type.");
+        }
+      }
+    }
 
-  @Override
-  public final void destroy() {
-    this.value.destroy();
-    this.destroyed = true;
-  }
+    @Override
+    public final void destroy() {
+      this.value.destroy();
```

Comment:
Set `this.value` to an empty `UnsignedByteArray`  so the underlying UBA can be GC'd

Reply:
This already happens because `UnsignedByteArray.destroy()` clears out its `List` of `UnsignedBytes`